src/main/java/com/uber/cadence/internal/replay/ClockDecisionContext.java

```diff
@@ -154,11 +187,93 @@ private void timerCancelled(long startEventId, Exception reason) {
     return result;
   }
 
+  /**
+   * @param id mutable side effect id
+   * @param func given the value from the last marker returns value to store. If result is empty
+   *     nothing is recorded into the history.
+   * @return the latest value returned by func
+   */
+  Optional<byte[]> mutableSideEffect(
+      String id,
+      Func1<MutableSideEffectData, byte[]> markerDataSerializer,
+      Func1<byte[], MutableSideEffectData> markerDataDeserializer,
+      Func1<Optional<byte[]>, Optional<byte[]>> func) {
+    MutableSideEffectResult result = mutableSideEffectResults.get(id);
+    Optional<byte[]> stored;
+    if (result == null) {
+      stored = Optional.empty();
+    } else {
+      stored = Optional.of(result.getData());
+    }
+    long eventId = decisions.getNextDecisionEventId();
+    int accessCount = result == null ? 0 : result.getAccessCount();
+    try {
```

Comment:
This try/catch is now not necessary. The internal code does not throw checked exception.